gooddata-sdk/gooddata_sdk/catalog/data_source/declarative_model/data_source.py

```diff
@@ -0,0 +1,102 @@
+# (C) 2022 GoodData Corporation
+from __future__ import annotations
+
+from typing import Any, Optional
+
+from gooddata_metadata_client.model.declarative_data_source import DeclarativeDataSource
+from gooddata_metadata_client.model.declarative_data_sources import DeclarativeDataSources
+from gooddata_sdk.catalog.data_source.declarative_model.physical_model.pdm import CatalogDeclarativePdm
+from gooddata_sdk.catalog.entity import CatalogTypeEntity
+
+
+class CatalogDeclarativeDataSources:
+    def __init__(self, data_sources: list[CatalogDeclarativeDataSource]):
+        self.data_sources = data_sources
+
+    @classmethod
+    def from_api(cls, entity: dict[str, Any]) -> CatalogDeclarativeDataSources:
+        data_sources = [CatalogDeclarativeDataSource.from_api(v) for v in entity["data_sources"]]
+        return cls(data_sources)
+
+    def to_api(self) -> DeclarativeDataSources:
+        return DeclarativeDataSources(data_sources=[v.to_api() for v in self.data_sources])
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, CatalogDeclarativeDataSources):
+            return False
+        return self.data_sources == other.data_sources
+
+
+class CatalogDeclarativeDataSource(CatalogTypeEntity):
+    def __init__(
+        self,
+        id: str,
+        type: str,
+        name: str,
+        url: str,
+        enable_caching: bool,
+        pdm: CatalogDeclarativePdm,
+        schema: Optional[str],
+        cache_path: Optional[list[str]] = None,
+    ):
+        super(CatalogDeclarativeDataSource, self).__init__(id, type)
+        self.schema = schema
+        self.enable_caching = enable_caching
+        self.cache_path = cache_path
+        self.name = name
+        self.url = url
+        self.pdm = pdm
+
+    @classmethod
+    def from_api(cls, entity: dict[str, Any]) -> CatalogDeclarativeDataSource:
+        return cls(
+            id=entity["id"],
+            name=entity["name"],
+            url=entity["url"],
+            type=entity["type"],
+            enable_caching=entity["enable_caching"],
+            pdm=CatalogDeclarativePdm.from_api(entity),
+            schema=entity.get("schema"),
```

Comment:
schema should be mandatory

Reply:
Hm. In the case of Dremio/Drill, it is not, but users must send an empty string, so you are right, it is mandatory.
We have to fix it in the future, now I switch it to mandatory.

Reply:
fixed